src/app/new-courses-page/new-courses-page.component.ts

```diff
@@ -8,15 +9,30 @@ import { Course } from '@app/models';
   templateUrl: './new-courses-page.component.html',
   styleUrls: ['./new-courses-page.component.scss']
 })
-export class NewCoursesPageComponent {
+export class NewCoursesPageComponent implements OnInit {
+  courseId: string;
+  course: Course;
 
-  constructor(private coursesService: CoursesService) {}
+  constructor(
+    private coursesService: CoursesService,
+    private router: Router,
+    private route: ActivatedRoute,
+  ) {}
+
+  ngOnInit(): void {
+    this.courseId = this.route.snapshot.paramMap.get('id');
+
+    if (this.courseId !== 'new') {
```

Comment:
You won't have this specific check when you switch route config places.
Use this check instead)
`this.courseId != null`

Reply:
Changed. Thanks =)